src/internal/interpreter/mod.rs

```diff
@@ -16,28 +21,29 @@ pub struct Interpreter {
     /// Holds temporary values
     stack: Vec<Constant>,
 
-    // Constant pool
-    constant_pool: Vec<Constant>,
+    code_holder: CodeHolder,
 
     /// Holds global variables
-    global: Vec<Option<Constant>>
+    global: Vec<Option<Constant>>,
 }
 
 impl Interpreter {
     /// Creates a new `Interpreter` instance
-    pub fn new() -> Interpreter {
+    pub fn new(ch: CodeHolder) -> Interpreter {
```

Comment:
I would highly advise not to add an argument to `new` as it's inconsistent with everything else with `new`. Instead, I suggest implementing `From` for Interpreter